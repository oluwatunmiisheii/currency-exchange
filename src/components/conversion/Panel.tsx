import React from 'react';
import Example from '../dropdown/Dropdown';
import Input from '../common/input/Input';
import { IInput } from '../../models/Input.model';

interface PanelProps extends IInput {
  children?: React.ReactNode;
}
 
const Panel: React.FC<PanelProps> = (props) => {
  return ( 

      <div>
        <div className="flex items-end">
          <div className="w-3/5">
            <Input
              hasLabel={props.hasLabel}
              label={props.label}
              inputId={props.inputId}
              type={props.type}
              name={props.name}
              placeholder={props.placeholder}
              disabled={props.disabled}
            />
          </div>
          <div className="w-2/5 ml-6">
            <Example />
          </div>
        </div>
        <>
          <h6 className="text-gray-500 text-sm inline-flex items-center">
            {props.children}
          </h6>
        </>
      </div>
  );
}
 
export default Panel;
