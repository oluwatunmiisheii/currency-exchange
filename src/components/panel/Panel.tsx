import React from 'react';
import Input from '../common/input/Input';
import { IInput } from '../../models/Input.model';
import CustomPopover from '../common/popover/CustomPopover';

interface PanelProps extends IInput {
  children?: React.ReactNode;
  popoverContent: React.ReactElement,
  currency: string,
  balance: number,
  errorMessage?: string,
}

const Panel: React.FC<PanelProps> = (props) => {
  return (
    <div>
      <div className="flex items-end">
        <div className="w-3/5">
          <Input
            hasLabel={props.hasLabel}
            label={props.label}
            inputId={props.inputId}
            type={props.type}
            name={props.name}
            placeholder={props.placeholder}
            disabled={props.disabled}
            onAmountChange={props.onAmountChange}
            value={props.value}
            errorMessage={props.errorMessage}
            hasError={props.hasError}
          />
        </div>
        <div className="w-2/5 ml-6">
          <CustomPopover currency={props.currency} balance={props.balance}>
            {props.popoverContent}
          </CustomPopover>
        </div>
      </div>
      <>
        {props.children}
      </>
    </div>
  );
}

export default Panel;
