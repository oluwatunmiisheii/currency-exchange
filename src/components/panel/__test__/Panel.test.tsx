import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import Panel, { IPanelProps } from '../Panel';

const onAmountChange = jest.fn();

const panelProps: IPanelProps = {
  inputId: 'test-input',
  label: 'test-label',
  name: 'test-name',
  type: 'text',
  placeholder: 'test-placeholder',
  value: 'test-value',
  onAmountChange: () => {},
  hasError: false,
  errorMessage: '',
  children: <div>test-children</div>,
  currency: 'EUR',
  balance: 0.00,
  popoverContent: <div>test-popover-content</div>,
};

const setup = (props: IPanelProps) => render(<Panel {...props} />);
 

describe('Panel', () => {
  it('should render custom popover component when trigger button is clicked', async () => {
    setup({...panelProps});
    
    const togglePopOverButton = screen.getByTestId('toggle-popover');
    fireEvent.click(togglePopOverButton);

    const popOverContent = screen.getByText(/^test-popover-content$/i); // full string match, ignore case
  
    expect(popOverContent).toBeInTheDocument();
    expect(popOverContent).toHaveTextContent('test-popover-content');

  });

  it('Should update input field when onAmountChange prop is Called', async () => {
    const { container } = setup({
      ...panelProps,
      onAmountChange,
      hasError: false,
      errorMessage: '',
    });
  
    const input = screen.getByTestId('test-input');
    fireEvent.change(input, { target: { value: '0.00' } });
  
    expect(onAmountChange).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  })

})
