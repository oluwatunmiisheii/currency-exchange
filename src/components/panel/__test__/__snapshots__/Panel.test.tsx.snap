// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel Should update input field when onAmountChange prop is Called 1`] = `
<div>
  <div>
    <div
      class="flex items-end"
    >
      <div
        class="w-3/5"
      >
        <div
          class="mt-1 border-b border-gray-300 focus-within:border-indigo-600 relative"
        >
          <input
            class="revolute-input"
            data-testid="test-input"
            id="test-input"
            name="test-name"
            placeholder="test-placeholder"
            type="text"
            value="test-value"
          />
        </div>
      </div>
      <div
        class="w-2/5 ml-6"
      >
        <button
          class="inline-flex justify-between w-full border-0 text-left border-b border-gray-300 px-4 py-2 text-sm text-gray-700 focus:outline-none"
          data-testid="toggle-popover"
        >
          <div>
            <div
              class="font-bold text-lg"
              data-testid=""
            >
              EUR
            </div>
            <div
              class="text-gray-500"
            >
              Balance: 
              â‚¬0.00
            </div>
          </div>
          <svg
            aria-hidden="true"
            class="h-5 w-5 text-gray-500"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
    <div>
      test-children
    </div>
  </div>
</div>
`;
